tests/x11/mate_terminal.pm

```diff
@@ -25,6 +25,8 @@ sub run {
     $self->enter_test_text('mate-terminal', cmd => 1);
     assert_screen 'test-mate_terminal-1';
     send_key "alt-f4";
+    assert_screen([qw(mate-terminal-confirm_close generic-desktop)]);
+    send_key 'ret' if match_has_tag 'mate-terminal-confirm_close';
```

Comment:
I'd argue that we'd want to recheck to have found a generic desktop after pressing 'RET' - to be sure the window is now really gone. In worst case it would stick around and cause issues in follow up tests otherwise

Reply:
yes, that happens implicitly in the post_run hook. Was like that in before and still is the case. In the case of the `generic-desktop` already found in L28 we are just checking it double but I guess that's ok.